packages/editor/src/components/post-author/hook.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+import { AUTHORS_QUERY, BASE_QUERY } from './constants';
+
+export function useAuthorsQuery( search ) {
+	const { authorId, authors, postAuthor } = useSelect(
+		( select ) => {
+			const { getUser, getUsers } = select( coreStore );
+			const { getEditedPostAttribute } = select( editorStore );
+			const _authorId = getEditedPostAttribute( 'author' );
+			const query = { ...AUTHORS_QUERY };
+
+			if ( search ) {
+				query.search = search;
+			}
+
+			return {
+				authorId: _authorId,
+				authors: getUsers( query ),
+				postAuthor: getUser( _authorId, BASE_QUERY ),
```

Comment:
The `query` argument must have a stable reference for `getEntityRecord` to return the [memoized value](https://github.com/WordPress/gutenberg/pull/34323). The issue is more noticeable when passing the `_fields` argument - the selector will return a new reference on each call.

cc @youknowriad

<details><summary>Unit Test</summary>
<p>

```javascript
	it( 'should return same reference for filtered item', () => {
		const state = deepFreeze( {
			entities: {
				records: {
					postType: {
						post: {
							queriedData: {
								items: {
									default: {
										1: {
											id: 1,
											content: 'chicken',
											author: 'bob',
										},
									},
								},
								itemIsComplete: {
									default: {
										1: true,
									},
								},
								queries: {},
							},
						},
					},
				},
			},
		} );
		expect(
			getEntityRecord( state, 'postType', 'post', 1, {
				_fields: 'author,content',
			} )
		).toBe(
			getEntityRecord( state, 'postType', 'post', 1, {
				_fields: 'author,content',
			} )
		);
	} );
``` 

</p>
</details>